run_tests.py

```diff
@@ -0,0 +1,52 @@
+import os
+import sys
+import contextlib
+import subprocess
+
+
+@contextlib.contextmanager
+def binding(binding):
+    """Prepare an environment for a specific binding"""
+
+    sys.stderr.write("""\
+#
+# Running tests with %s..
+#
+""" % binding)
+
+    os.environ["QT_PREFERRED_BINDING"] = binding
+
+    try:
+        yield
+    except:
+        pass
+
+    os.environ.pop("QT_PREFERRED_BINDING")
+
+
+if __name__ == "__main__":
+    argv = [
+        "nosetests",
+        "--verbose",
+        "--with-process-isolation",
+        "--exe",
+    ]
+
+    argv.extend(sys.argv[1:])
+
+    # Running each test independently via subprocess
+    # enables tests to filter out from tests.py before
+    # being split into individual processes via the
+    # --with-process-isolation feature of nose.
+    with binding("PyQt4"):
+        subprocess.call(argv)
```

Comment:
Sorry missed this comment.

Cons, not that I can tell. There are some organisational cons; as we are now doing in Python what we could have done in the docker ENTRYPOINT, and in the ENTRYPOINT we are doing things that could have been done here.

I sense that at some point we'll either transition into doing it all there, or all in Python. But personally I feel it's too early to tell and would be happy to leave it as-is until we can point at concrete problems and manage them then. Unless you see any issues with it as-is?